docs/usage/self-hosted-experimental.md

```diff
@@ -1,22 +1,23 @@
-# Self-Hosted Experimental Options
+# Self-hosted experimental options
 
-The following options are considered "experimental" in that:
+The following options are "experimental" because:
 
-- They should not commonly be needed
-- They are typically an effort to work around some other service or platform's problem
-- We hope they can each be removed one day, or we are waiting to be sure they work as intended
+- They are not commonly needed
+- They are typically an effort to work around some other service's or platform's problem
+- They can be removed at any time
+- They are options for Renovate's internal use to validate they work as intended
 
-Experimental options which are commonly used and for which there is no external solution in sight should be replaced with official options.
+Experimental options which are commonly used and for which there is no external solution in sight should be replaced with official options ("Who is doing the replacing here? Renovate bot team or end-user?").
 
-Please use experimental options at your own risk - they may be removed or behavior changed in any release, not just major, however all attempts will be made not to inconvenience anyone.
+Use experimental options at your own risk - they may be removed or behavior changed in any release, not just major, however all attempts will be made not to inconvenience anyone. (Aren't we breaking our "We always follow SemVer promise here?")
```

Comment:
## Relevant SemVer rules:

>5. Version 1.0.0 defines the public API. The way in which the version number is incremented after this release is dependent on this public API and how it changes.

> 7. Minor version Y (x.Y.z | x > 0) MUST be incremented if new, backwards compatible functionality is introduced to the public API. It MUST be incremented if any public API functionality is marked as deprecated. It MAY be incremented if substantial new functionality or improvements are introduced within the private code. It MAY include patch level changes. Patch version MUST be reset to 0 when minor version is incremented.

> 8. Major version X (X.y.z | X > 0) MUST be incremented if any backwards incompatible changes are introduced to the public API. It MAY also include minor and patch level changes. Patch and minor version MUST be reset to 0 when major version is incremented.

## Why I think we're breaking our SemVer promise

I think that having these experimental options is still a "public API", even if we "hide it away" in the code and docs, and warn our users away from using them. So shouldn't we follow SemVer still?

The way I read it the lifetime of a experimental option should be:

1. Introduce experimental option in a `minor` version, as "Minor version Y (x.Y.z | x > 0) MUST be incremented if new, backwards compatible functionality is introduced to the public API."
2. Deprecate experimental option in a `minor` version as "Minor version MUST be incremented if any public API is marked as deprecated".
3. Remove option completely in a `major` version, as we introduce a "backwards incompatible change to the public API".

Reply:
No, because they are explicit experimental and therefore don't follow semver. But we try to avoid breaking changes on non-major if possible.

Reply:
Maybe we should be a bit clearer that we're not following SemVer for the experimental options:

```suggestion
Use experimental options at your own risk.
We do not follow Semantic Versioning for any experimental option.
These options may be removed or have their behavior changed in **any** version.
We will try to keep breakage to a minimum, but make no guarantees that a experimental option will keep working.
```